components/sup/src/manager/service/spec.rs

```diff
@@ -124,7 +124,12 @@ impl IntoServiceSpec for protocol::ctl::SvcLoad {
             spec.update_strategy = UpdateStrategy::from_i32(update_strategy).unwrap_or_default();
         }
         if let Some(ref list) = self.binds {
-            spec.binds = list.binds.clone().into_iter().map(Into::into).collect();
+            spec.binds = list
+                .binds
+                .clone()
+                .into_iter()
```

Comment:
Both the `clone` and `into_iter` can be omitted in favor of a simple `iter`. We only need to clone the `service_group` field, and we can do it like this:
```rust
                .iter()
                .map(|b| ServiceBind::new(&b.name, b.service_group.clone().into()))
```
Note, I also removed the superfluous `hcore::service::` since there's already a
```rust
use crate::hcore::service::{…, ServiceBind};
```
in this file. If it was intentional to show the type conversion, I think it's more illustrative to fully qualify *both* types:
```rust
                .map(|pb: &habitat_sup_protocol::types::ServiceBind| {
                    hcore::service::ServiceBind::new(&pb.name, pb.service_group.clone().into())
                })
```

Reply:
Yeah, it was to show the type conversion.  Good idea on showing both types